internal/pkg/config/v1/config.go

```diff
@@ -259,14 +266,16 @@ func (c *Config) encryptContextStateTokens(tempAuthToken, tempAuthRefreshToken s
 		c.Context().GetState().Salt = salt
 		c.Context().GetState().Nonce = nonce
 	}
-	if tempAuthToken != "" {
+
+	if tempAuthToken != "" && regexp.MustCompile(authTokenRegex).MatchString(tempAuthToken) {
 		encryptedAuthToken, err := secret.Encrypt(c.Context().Name, tempAuthToken, c.Context().GetState().Salt, c.Context().GetState().Nonce)
 		if err != nil {
 			return err
 		}
 		c.Context().GetState().AuthToken = encryptedAuthToken
 	}
-	if tempAuthRefreshToken != "" {
+
+	if tempAuthRefreshToken != "" && regexp.MustCompile(authRefreshTokenRegex).MatchString(tempAuthRefreshToken) {
```

Comment:
This regular expression does not match the empty string, so this is redundant.
```suggestion
	if regexp.MustCompile(authRefreshTokenRegex).MatchString(tempAuthRefreshToken) {
```